src/utils/hooks/use-generate-pathname.js

```diff
@@ -0,0 +1,49 @@
+import { useCallback } from 'react';
+import { useHistory, generatePath } from 'react-router-dom';
+import { routes } from '../../routes.config';
+import { localStorageName } from '../../config';
+
+/**
+ * To generate different pathnames based on each action
+ * E.g.: click on a node, or focus on a modular pipeline
+ * or to reset the pathname to the main page
+ */
+export const useGeneratePathname = () => {
+  const history = useHistory();
+
+  const localStorage = window.localStorage.getItem(localStorageName);
+  const activePipeline = JSON.parse(localStorage)?.pipeline?.active;
+
+  const toFlowchartPage = useCallback(() => {
+    const url = generatePath(routes.flowchart.main);
+    history.push(url);
+  }, [history]);
+
+  const toSelectedNode = useCallback(
```

Comment:
I think I mentioned this is the demo on Friday: could we also have a function here much like these other three called something like `toSelectedPipeline` where we only add the `pipelineId` to the path and then get something like this on the frontend?

<img width="510" alt="image" src="https://user-images.githubusercontent.com/16869061/197503741-4170d9fd-ebfd-4ec2-9055-1b46e564b124.png">

Or is this what you were saying about us always needing **both** the `pipelineId` and `id` in the URL?

Reply:
hey yeah i think we need both in the URL as we're using query parameter. So if you just update the pipelineId, the URL will miss the other id `http://localhost:4141/?pipeline_id=Modelling%20stage&selected_id=`

Unless theres away to show `&selected_id=` conditionally in the URL